taskfile/read/cache.go

```diff
@@ -0,0 +1,59 @@
+package read
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type Cache struct {
+	dir string
+}
+
+func NewCache(dir string) (*Cache, error) {
+	dir = filepath.Join(dir, "remote")
+	if err := os.MkdirAll(dir, 0o755); err != nil {
+		return nil, err
+	}
+	return &Cache{
+		dir: dir,
+	}, nil
+}
+
+func checksum(b []byte) string {
+	h := sha256.New()
+	h.Write(b)
+	return base64.StdEncoding.EncodeToString(h.Sum(nil))
```

Comment:
Almost there. We want the just the sha265 and don't need to encode it as base64:

```suggestion
	return fmt.Sprintf("%x", h.Sum(nil))
```

This was even causing a side-effect bug for me:

```
The task you are attempting to run depends on the remote Taskfile at "https://gist.githubusercontent.com/andreynering/8a786860ce88defcaea5bb0af22cd984/raw/552901beadae8f09ed2bfffe256f5db877cd7fcc/RemoteTaskfile.yml".
--- Make sure you trust the source of this Taskfile before continuing ---
Continue? [y/N]
y
open /Users/andrey/Developer/andrey/task/tmp/.task/remote/vRma/iPJM55JsckfKEXuArHXXHF4Gs/5xPGRgceeWao.checksum: no such file or directory
```

Reply:
@andreynering Ah, I meant to use `URLEncoding` here instead which would also fix the issue. Hex is fine though, just a little bit longer. I will be away this weekend, but please feel free to merge this if you think it's ready.